canonical/snapcraft.io
# feat: add unregister snap name logic

## Done
Adds Unregister snap name logic to `/snaps` page and to the Settings tab of a registered snap.

## How to QA
- Check visuals on [`snaps`](https://snapcraft-io-4670.demos.haus/snaps) page match design [here](https://app.zeplin.io/project/651a8c4cdfae28202eb7c7c6/screen/651a8c8669cd5020b890099b)
- Check visuals on Settings tab match design [here](https://app.zeplin.io/project/651a8c4cdfae28202eb7c7c6/screen/651a8c86537a12201042e75e)
- Check 'Unregister' button is disabled on `/snaps` page as well as on a snap you are not a publisher of

***If you have no registered snap names, check the videos below.***

Otherwise:
1. If you have a registered snap name that you have not published, go to [`/snaps`](https://snapcraft-io-4670.demos.haus/snaps) and try to unregister it.
2. Go to the Settings tab of a registered snap name and unregister from here.


## Testing

## Issue / Card
Fixes https://warthogs.atlassian.net/browse/WD-6771

## Screenshots

Successfully unregister a snap name from `/snaps`:
[Screencast from 03-06-24 09:23:57.webm](https://github.com/canonical/snapcraft.io/assets/74302970/577969c6-5238-42f8-ae00-61bbc3b65152)

Successfully unregister a snap name from Settings tab, and reload page to `/snaps`
[Screencast from 03-06-24 09:24:45.webm](https://github.com/canonical/snapcraft.io/assets/74302970/81e8bea1-a5db-49df-b265-fd1669053706)

Unregister button disabled when I am not a publisher:
![Screenshot from 2024-06-04 10-33-40](https://github.com/canonical/snapcraft.io/assets/74302970/4caf6c58-6cfe-42e4-b421-2c114f7ed87e)
![Screenshot from 2024-06-04 10-33-59](https://github.com/canonical/snapcraft.io/assets/74302970/0b78cf33-3dfc-414f-a607-61e88c738ad1)
